Makefile

```diff
@@ -147,9 +148,15 @@ tests-privileged:
 	$(QUIET) $(MAKE) $(SUBMAKEOPTS) -C bpf cilium-map-migrate
 	$(MAKE) init-coverage
 	for pkg in $(patsubst %,github.com/cilium/cilium/%,$(PRIV_TEST_PKGS)); do \
-		PATH=$(PATH):$(ROOT_DIR)/bpf $(GO_TEST) $(TEST_LDFLAGS) $$pkg $(GOTEST_UNIT_BASE) $(GOTEST_COVER_OPTS) \
-		|| exit 1; \
-		tail -n +2 coverage.out >> coverage-all-tmp.out; \
+		if [ "${RACE}" == "" ]; then \
+			PATH=$(PATH):$(ROOT_DIR)/bpf $(GO_TEST) $(TEST_LDFLAGS) $$pkg $(GOTEST_UNIT_BASE) $(GOTEST_COVER_OPTS) \
+			|| exit 1; \
+			tail -n +2 coverage.out >> coverage-all-tmp.out; \
+		else \
+			PATH=$(PATH):$(ROOT_DIR)/bpf go test -mod=vendor -race -tags lockdebug,privileged_tests $(TEST_LDFLAGS) $$pkg $(GOTEST_PRIV_OPTS) $(GOTEST_COVER_OPTS) \
```

Comment:
can we use `$(GO_TEST)` here too please?

Reply:
@kAworu we can, can you derive a good one? I _think_ we don't even need to use the `if` condition has I suggested to Maciej on line 151 if we properly do it in `Makefile.defs`

Reply:
@aanm the `-race`, `-mod=vendor` and `-tags` flags are already handled by Makefile.defs when `$(GO_TEST)` is used. The only thing that stand out to me is that in `$(GO_TEST)` the `lockdebug` tag is only provided when `$(LOCKDEBUG)` is enabled, where here is it provided unconditionally.

Beside that,  I see a difference in other arguments (specifically `$(GOTEST_UNIT_BASE)` vs `$(GOTEST_PRIV_OPTS)`), so that should be sorted out in order to remove the `if` condition correct?

Reply:
I don't see the harm of setting, if not set, `LOCKDEBUG=1` when `RACE=1`.

@kAworu correct!

Reply:
So basically, get rid of the `if` and set `LOCKDEBUG=1` if `RACE` is set? Pushing the change soon.